src/reanimated2/Colors.ts

```diff
@@ -462,14 +462,16 @@ export const rgbaColor = (
  */
 export function RGBtoHSV(rgb: RGB): HSV;
 export function RGBtoHSV(r: number, g: number, b: number): HSV;
-export function RGBtoHSV(r: any, g?: any, b?: any): HSV {
+export function RGBtoHSV(r: RGB | number, g?: number, b?: number): HSV {
   'worklet';
-  /* eslint-disable */
-  if (arguments.length === 1) {
+  if (typeof r === 'object') {
     g = r.g;
     b = r.b;
     r = r.r;
   }
+  // Typescript doesn't know that g and b are now sure to not be undefined
```

Comment:
Can we somehow achieve this using function overloading?

Reply:
Actually took the time to profile those and more solutions, we will make a decision later on how to handle problems with types in function overloads.